### PR TITLE
Mutability documentation and schema definition

### PR DESCRIPTION
**Pull request by amarzavery**

### CODE REVIEW

File: schema/swagger-extensions.json [hunk 1]
```diff
@@ -1578,6 +1581,14 @@
...
       "type": "string",
       "enum": ["client", "method"]
+    },
+    "xmsMutability": {
+      "type": "array",
+      "items": {
+        "type": "string",
+        "enum": ["create, read", "update"]
+      },
+      "uniqueItems": true
```
Comment (amarzavery) [inline schema/swagger-extensions.json]: @tbombach - One less linter rule to assert that the extension should have unique values.

**Comment by amarzavery** [comment]
In case you want to see the markdown file directly https://github.com/amarzavery/AutoRest/blob/mutability/Documentation/swagger-extensions.md#x-ms-mutability

File: Documentation/swagger-extensions.md [hunk 2]
```diff
@@ -15,6 +15,7 @@ The following documents describes AutoRest specific vendor extensions for [Swagg
 * [x-ms-discriminator-value](#x-ms-discriminator-value) - maps discriminator value on the wire with the definition name.
 * [x-ms-client-flatten](#x-ms-client-flatten) - flattens client model property or parameter.
 * [x-ms-parameterized-host](#x-ms-parameterized-host) - replaces the Swagger host with a host template that can be replaced with variable parameters.
+* [x-ms-mutability](#x-ms-mutability) - provides insight to Autorest on how to generate code. It doesn't alter the modeling of what is actually sent on the wire.
```
Comment (kirthik) [inline Documentation/swagger-extensions.md]: @amarzavery : Can we add this documentation in rest-api-specs repo? We decided to add swagger related documentation in that repository right?

Comment (amarzavery) [inline Documentation/swagger-extensions.md]: sure. I will send a PR over there as well. I wanted to make sure that the description and the schema changes are both reviewed in one place.

Does everything else - documentation and schema changes look correct?

File: schema/swagger-extensions.json [hunk 3]
```diff
@@ -1578,6 +1581,14 @@
     "xmsParameterLocation": {
       "type": "string",
       "enum": ["client", "method"]
+    },
+    "xmsMutability": {
+      "type": "array",
+      "items": {
+        "type": "string",
+        "enum": ["create, read", "update"]
```
Comment (kirthik) [inline schema/swagger-extensions.json]: Shouldn't this be `["create","read","update"]` (end double quotes after create and new quotes for read)

Comment (amarzavery) [inline schema/swagger-extensions.json]: yup. thanks for pointing that.

**Comment by fearthecowboy** [comment]
LGTM 👍

**Comment by tbombach** [comment]
LGTM as well

**Pull request merged**